### DISCUSSION TITLE
Upgrade VSIX Project w/Tool Window (Community) to Framework 5.0

### DISCUSSION THREAD

**Discussion by habaneando** [VsixCommunity/Community.VisualStudio.Toolkit#233] [Q&A] [answered]
I'm working with VSIX Project w/Tool Window (Community) template using VS2019 & [Extensibility Essentials 2019](https://marketplace.visualstudio.com/items?itemName=MadsKristensen.ExtensibilityEssentials2019)
It only supports Framework 4.7.2 or minor versions but then I want to use libraries from 5.0 to perform code generation. 
I would not like to downgrade the libraries to 4.7.2 version.
Do I need to switch to VS2022 & [Extensibility Essentials 2022](https://marketplace.visualstudio.com/items?itemName=MadsKristensen.ExtensibilityEssentials2022) in order to get the vsix project template working with 5.0 or can I take another workaround and keep using the VS2019?

**Comment by StevenRasmussen** [comment, marked as answer]
Unfortunately, the toolkit will not be able to support .Net 5+ until Visual Studio itself targets it.  See also #56.  The only option you really have is to run your code out-of-process... meaning via a command line or something of that nature.

**Comment by habaneando** [comment]
Hi Steven,
I have been using code generation (Roslyn APIs) from Console during some time ago. 
Recently I created my custom solution project template from scratch to follow some specific architecture design.
I also created a Tool Window to call my code generation using some inputs directly from Solution Explorer.
That's why I'm missing the last piece of the puzzle.
I wish my team can run the code generation directly from Solution Explorer in VS without open another solution like they have been doing now.  
It is quite weird VS2022 still using the VSIX Project template with .Net Framework 4.8.
Maybe I should start thinking into downgrade my code to 4.7.2 in order to use it into the extension. 
Anyways you are totally right. It looks like it is not an option for now. 
Thanks a lot for your quick and useful reply.
